src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageProcessor.java

```diff
@@ -0,0 +1,158 @@
+     */
+    public boolean processNextMessage() throws ServiceBusException, InterruptedException {
+        IMessage message = messageReceiver.receive();
+        if (message != null) {
+            try {
+                log.info("Started processing notification message with ID {}", message.getMessageId());
+                var notificationMsg = notificationMessageParser.parse(message.getMessageBody());
+                notificationMessageHandler.handleNotificationMessage(notificationMsg);
+                finaliseProcessedMessage(message, MessageProcessingResult.SUCCESS);
+            } catch (InvalidMessageException ex) {
+                log.error("Invalid notification message with ID: {} ", message.getMessageId(), ex);
+                finaliseProcessedMessage(message, MessageProcessingResult.UNRECOVERABLE_FAILURE);
+            } catch (Exception ex) {
+                log.error("Failed to process notification message with ID: {} ", message.getMessageId(), ex);
+                finaliseProcessedMessage(message, MessageProcessingResult.POTENTIALLY_RECOVERABLE_FAILURE);
+            }
+        } else {
+            log.debug("No notification messages to process by notification processor.");
+        }
+
+        return message != null;
+    }
+
+
+    private void finaliseProcessedMessage(IMessage message, MessageProcessingResult processingResult) {
+        try {
+            completeProcessedMessage(message, processingResult);
+        } catch (InterruptedException ex) {
+            log.error(
+                "Failed to finalise notification message with ID {}. Processing result: {}",
+                message.getMessageId(),
+                processingResult,
+                ex
+            );
+            Thread.currentThread().interrupt();
+        } catch (Exception ex) {
+            log.error(
+                "Failed to finalise notification message with ID {}. Processing result: {}",
+                message.getMessageId(),
+                processingResult,
+                ex
+            );
+        }
+    }
+
+    private void completeProcessedMessage(
+        IMessage message,
+        MessageProcessingResult processingResult
+    ) throws InterruptedException, ServiceBusException {
+
+        switch (processingResult) {
+            case SUCCESS:
+                messageReceiver.complete(message.getLockToken());
+                log.info("Notification Message with ID {} has been completed successfully.", message.getMessageId());
+                break;
+            case UNRECOVERABLE_FAILURE:
+                deadLetterTheMessage(
+                    message,
+                    "Notification Message processing error",
+                    "UNRECOVERABLE_FAILURE"
+                );
+                break;
+            case POTENTIALLY_RECOVERABLE_FAILURE:
+                deadLetterIfMaxDeliveryCountIsReached(message);
+                break;
+            default:
```

Comment:
This code is dead because there are no other possible values for processingResult. Makes sense to use if - else if - else construction

Reply:
if we add new result type  and forget to update here it would give better error. 
but we can remove the default and exception.

Reply:
I think if you remove it, it won't pass quality gate.

Reply:
Iyou are right